### ISSUE TITLE
Support Regex in module value, function in rename value when selecting `module`

### ISSUE THREAD

**Issue by tjcouch-sil** [UmamiAppearance/rollup-plugin-import-manager#54]
Hi! 👋 
      
Firstly, thanks for your work on this project! 🙂

Today I used [patch-package](https://github.com/ds300/patch-package) to patch `rollup-plugin-import-manager@0.5.3` for the project I'm working on [source code for the plugin configuration here](https://github.com/paranext/paranext-extension-template/blob/94fd341d4d4c80e1fb41122ca425dfc2b317da19/vite/vite.config.ts#L39).

I needed to make exact matches to module names (instead of the current matching where it looks for the config-provided module name anywhere in the module name string) so that I could remove a specific module but not modules whose names included that module name, so I added Regex support in the module name.

I also needed to modify the relative path to modules programmatically when renaming them, not only replace module names one-for-one, so I added the ability to provide a function that receives the full raw module source path and returns a new full raw module source path.

Here is the diff that solved my problem:

```diff
diff --git a/node_modules/rollup-plugin-import-manager/README.md b/node_modules/rollup-plugin-import-manager/README.md
index f1bbebd..b13192a 100644
--- a/node_modules/rollup-plugin-import-manager/README.md
+++ b/node_modules/rollup-plugin-import-manager/README.md
@@ -162,7 +162,7 @@ This is where the plugin comes to life. Here is the place where units are gettin
 ---
 
 #### `module` <samp>[option for units]</samp>
-Type: `String`  
+Type: `String` | `RexExp`  
 Default: `null`  
 
 Selects a unit by its module name. Each import has a name object. This is constructed from the module.
@@ -353,10 +353,10 @@ An option to target an alias of a [selected](#select-option-for-actions) `defaul
 
 
 ##### `rename` <samp>[option for actions]</samp>
-Type: `String`  
+Type: `String` | `(moduleSourceRaw: string) => string`  
 Default: `null`  
 
-This option is used to rename a [selected](#select-option-for-actions) specific part (`defaultMember`, `member`, `module`). The value is the new name of the selected part. See this [example](#changing-the-module).
+This option is used to rename a [selected](#select-option-for-actions) specific part (`defaultMember`, `member`, `module`). The value is a string of the new name of the selected part. See this [example](#changing-the-module). If the selected part is `module`, the value could alternatively be a function accepting the whole raw module source (ex: `"./module-name"` including quotes) and returning the whole new name (ex: `"./newmodule-name"` including quotes).
 
 
 ##### `modType` <samp>[option for actions]</samp>
```

This change corresponds to https://github.com/UmamiAppearance/ImportManager/issues/47

Please let me know if you would like me to submit a pull request with these changes.

<em>This issue body was [partially generated by patch-package](https://github.com/ds300/patch-package/issues/296).</em>

**Comment by UmamiAppearance** [comment]
Hi.
This is a great idea. Regex support would enhance module selection by a lot. Please submit a pull request. I have already created a new branch:
[regex-module-selection](https://github.com/UmamiAppearance/rollup-plugin-import-manager/tree/regex-module-selection)

**Comment by tjcouch-sil** [comment]
Created pull request https://github.com/UmamiAppearance/rollup-plugin-import-manager/pull/55

**Comment by UmamiAppearance** [comment]
I think we are good to go. I additionally introduced a method to [match the module via the raw source](https://github.com/UmamiAppearance/rollup-plugin-import-manager/tree/regex-module-selection_plugin#rawmodule-option-for-units).  
Your example has found its place [here](https://github.com/UmamiAppearance/rollup-plugin-import-manager/tree/regex-module-selection_plugin#changing-a-relative-path-to-different-directory-making-use-of-a-rename-function). It is narrowed down to the particularities of using a function.  
RegExp  has its own example [here](https://github.com/UmamiAppearance/rollup-plugin-import-manager/tree/regex-module-selection_plugin#applying-regexp-for-module-matching). 

Any final objections?

**Comment by tjcouch-sil** [comment]
I think that's great!! Thanks for the great support and help. The ImportManager dependency needs to be updated whenever you publish that first, though, right?

**Comment by UmamiAppearance** [comment]
Thanks for your contribution. This is a real enhancement. 👍 

> The ImportManager dependency needs to be updated whenever you publish that first, though, right?

Yes.

**Comment by UmamiAppearance** [comment]
Resolved in #56
